### PR TITLE
refactor(run-protocol): DRY checkDebtLimit

### PR DESCRIPTION
**Pull request by turadg**
Extracted from https://github.com/Agoric/agoric-sdk/pull/5135

## Description

- factor out `checkDebtLimit` that was copy-pasted
- more idiomatic assert
- more accurate error message

### Security Considerations

--

### Documentation Considerations

--

### Testing Considerations

--

### CODE REVIEW

File: packages/run-protocol/src/contractSupport.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // @ts-check
+import { q } from '@agoric/assert';
```
Comment (dckc) [inline packages/run-protocol/src/contractSupport.js]: I don't think I should approve code with this idiom. Your point that the deprecation isn't documented is well made, but I still think it's the position of several in the team, and as a reviewer, I'm obliged to represent them.

Comment (turadg) [reply packages/run-protocol/src/contractSupport.js]: Deferring resolution, moved to chat

**Review by dtribble — APPROVED** [review]
LGTM

**Review by dckc — APPROVED** [review]
good stuff.

I gather @dtribble is fine with this too.

**Pull request merged**